src/NRediSearch/Suggestion.cs

```diff
@@ -0,0 +1,108 @@
+﻿// .NET port of https://github.com/RedisLabs/JRediSearch/
+using System;
+using StackExchange.Redis;
+
+namespace NRediSearch
+{
+    public class Suggestion
+    {
+        public string String { get; }
+        public double Score { get; }
+        public string Payload { get; }
+
+        private Suggestion(SuggestionBuilder builder)
+        {
+            String = builder._string;
+            Score = builder._score;
+            Payload = builder._payload;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (this == obj)
+            {
+                return true;
+            }
+
+            if (obj == null || obj.GetType() != GetType())
+            {
+                return false;
+            }
+
+            var that = obj as Suggestion;
+
+            return Score.Equals(that.Score) && String.Equals(that.String) && Payload.Equals(that.Payload);
```

Comment:
`.Equals` here is dangerous re nullability; are they known to be not null?

Reply:
😬 Point taken. 

To answer your question, the builder would ensure that the value of `String` was set, but not that the value of `Payload`. 

Updating these.